interaction-manager/src/lib_components/touch-gestures.js

```diff
@@ -0,0 +1,340 @@
+/* eslint-disable max-lines-per-function */
+
+import { getDirection } from '../utils/gesture-utils';
+import { distanceBetweenTwoPoints, toDeg } from '../utils/utility-functions';
+
+/**
+ * TouchGestures class that handles all of the touch interactions and gestures
+ */
+class TouchGestures {
+    /* eslint-disable-next-line require-jsdoc */
+    constructor() {
+        this.activeTouches = new Map();
+    }
+
+    /**
+     *
+     * @param {Object} options
+     * @param {HTMLElement | string} options.element - Element you want to attach the touch event to
+     * @param {function} options.callback - Function to be executed on touch
+     * @param {number} [options.time=1000] - Time in milliseconds for the press
+     * @returns {void}
+     */
+    hold(options) {
+        if (!options) return console.error('Options not provided for hold!');
+
+        let holdTimer;
+
+        const element =
+            options.element instanceof HTMLElement ? options.element : document.querySelector(options.element);
+
+        if (!element) return console.error('Element not found!');
+
+        element.addEventListener('touchstart', ({ touches }) => {
+            this.activeTouches.set(touches[0].identifier, touches[0]);
+
+            clearTimeout(holdTimer);
+            holdTimer = setTimeout(() => {
+                if (!options.callback) return;
+                options.callback();
+            }, options.time || 1000);
+        });
+
+        element.addEventListener('touchend', ({ touches }) => {
+            this.activeTouches.delete(touches[0].identifier);
+            clearTimeout(holdTimer);
+        });
+    }
+
+    /**
+     *
+     * @param {Object} options
+     * @param {HTMLElement | string} options.element - Element you want to attach the touch event to
+     * @param {function} options.callback - Function to be executed on touch
+     * @param {number} [options.tapsNumber=1] - Number of taps necessary for the callback to be executed
+     * @returns {void}
+     */
+    tap(options) {
+        if (!options) return console.error('Options not provided for tap!');
+
+        let tapTimer, betweenTapsTimer;
+        let isTap = true;
+        let isNextTapReady = true;
+        let tapCount = options.tapsNumber || 1;
+
+        const element =
+            options.element instanceof HTMLElement ? options.element : document.querySelector(options.element);
+
+        if (!element) return console.error('Element not found!');

```

Comment:
It's nice, but there is a possibility of adding a HTMLElement object directly, instead of the selector. Meaning that users will get `Failed to find element [Object object] trying to tap` :(